sklearn/preprocessing/data.py

```diff
@@ -250,13 +313,65 @@ def fit(self, X, y=None):
         if feature_range[0] >= feature_range[1]:
             raise ValueError("Minimum of desired feature range must be smaller"
                              " than maximum. Got %s." % str(feature_range))
-        data_min = np.min(X, axis=0)
-        data_range = np.max(X, axis=0) - data_min
-        data_range = _handle_zeros_in_scale(data_range)
-        self.scale_ = (feature_range[1] - feature_range[0]) / data_range
-        self.min_ = feature_range[0] - data_min * self.scale_
-        self.data_range = data_range
-        self.data_min = data_min
+
+        if sparse.issparse(X):
+            raise TypeError("MinMaxScaler does no support sparse input.")
```

Comment:
Actually the `check_array` check at the beginning of this function will already reject sparse input because there is not `accept_sparse` argument. Still I thing it's a good idea to suggest to use an alternative scaler such as `MaxAbsScaler` so you should move the if `sparse.issparse(X)` test at the beginning of the method, before the `check_array` check.